pyclam/chaoda.py

```diff
@@ -0,0 +1,148 @@
+from typing import List, Dict, Set
+
+import numpy as np
+from scipy.special import erf
+
+from pyclam import Graph, Cluster
+
+_NORMALIZATION_MODES = ['linear', 'gaussian', 'sigmoid']
+
+
+def _normalize(scores: np.array, mode: str) -> np.array:
+    if mode == 'linear':
+        min_v, max_v, = float(np.min(scores)), float(np.max(scores))
+        if min_v == max_v:
+            max_v += 1.
+        else:
+            pass
+        scores = (scores - min_v) / (max_v - min_v)
+    else:
+        mu: float = float(np.mean(scores))
+        sigma: float = max(float(np.std(scores)), 1e-3)
+
+        if mode == 'gaussian':
+            scores = erf((scores - mu) / (sigma * np.sqrt(2)))
+        elif mode == 'sigmoid':
+            scores = 1 / (1 + np.exp(-(scores - mu) / sigma))
+        else:
+            raise ValueError(f'normalization mode {mode} is undefined. Use one of {_NORMALIZATION_MODES}.')
+
+    return scores.ravel().clip(0, 1)
+
+
+def cluster_cardinality(graph: Graph, normalize: str = 'gaussian') -> np.array:
```

Comment:
docstring should provide available normalization modes

Reply:
fixed